services/api/service.go

```diff
@@ -170,6 +190,15 @@ type RelayAPI struct {
 
 	payloadAttributes     map[string]payloadAttributesHelper // key:parentBlockHash
 	payloadAttributesLock sync.RWMutex
+
+	// The slot we are currently optimistically simulating.
+	optimisticSlot uint64
+	// The number of optimistic blocks being processed (only used for logging).
+	optimisticBlocksInFlight uint64
```

Comment:
should this not be an atomic? i.e. `uberatomic.Uint64`

Reply:
ya that is cleaner! done